ClassTypes/PowerShell/EthicalHacking/Setup-EthicalHacking.ps1

```diff
@@ -276,36 +279,44 @@ function Convert-VdmkToVhdx {
 Downloads Kali Linux Hyper-V image and returns the path to the virtual machine configuration information file.
 #>
 function Get-KaliLinuxDisk {
-    $vhdxPath = "$env:PUBLIC\Documents\Hyper-V\Virtual hard disks\Kali-Linux-2020.4-vmware-amd64.vhdx"
+    $_kaliLinuxVersion = '2023.3'
+    $_kaliLinuxBaseFileName = "kali-linux-$($_kaliLinuxVersion)a-hyperv-amd64"
+    $_kaliLinuxVhdxFileName = "$($_kaliLinuxBaseFileName).vhdx"
+
+    $_kaliLinuxExtractedFilesFolder = "$env:PUBLIC\Documents\Hyper-V\Virtual hard disks"
+    New-Item -Path $_kaliLinuxExtractedFilesFolder -ItemType Directory -Force | Out-Null
+   
+    $vhdxPath = "$env:PUBLIC\Documents\Hyper-V\Virtual hard disks\$_kaliLinuxVhdxFileName"
 
     if (Test-Path ($vhdxPath)) { return $vhdxPath }
 
     Write-Host "Downloading Kali Linux image compressed file. "
-     $_kaliLinuxExtractedFilesFolder = Join-Path $env:TEMP 'KaliLinux'
-    #Download page is https://www.offensive-security.com/kali-linux-vm-vmware-virtualbox-image-download/
-    $_kaliDownloadedFileName = 'kali-linux-2021.4-vmware-amd64.7z'
-    $kaliLinux7ZipFile = Get-WebFile -DownloadUrl 'https://kali.download/virtual-images/kali-2021.4/kali-linux-2021.4-vmware-amd64.7z' -TargetFilePath $(Join-Path $_kaliLinuxExtractedFilesFolder $_kaliDownloadedFileName) -SkipIfAlreadyExists $true
+
+
+    #Download page is https://www.kali.org/get-kali/#kali-virtual-machines
+    #Docs at https://www.kali.org/docs/virtualization/import-premade-hyperv/
+    $_kaliDownloadedFileName = "$_kaliLinuxBaseFileName.7z"
+    $kaliLinux7ZipFile = Get-WebFile -DownloadUrl "https://cdimage.kali.org/kali-$_kaliLinuxVersion/$_kaliDownloadedFileName" -TargetFilePath $(Join-Path $_kaliLinuxExtractedFilesFolder $_kaliDownloadedFileName) -SkipIfAlreadyExists $true
  
     $sevenZipExe = Join-Path $env:ProgramFiles '7-zip\7z.exe'
     if (-not (Test-Path $sevenZipExe)) {
         Write-Host "Downloading and installing 7-Zip to extract Kali Linux compressed files."
-          #Download page is https://www.7-zip.org/download.html.
+        #Download page is https://www.7-zip.org/download.html.
```

Comment:
You might want to write out to the console that the license for 7zip is here:  https://www.7-zip.org/license.txt and by using the script they are agreeing to the terms...

Reply:
Good point.  I added a call the `ShouldContinue` whenever script downloads files or installs software.  Also added 'Force' is someone wants to avoid prompts.